### PR TITLE
Add Integration spec

### PR DESCRIPTION
**Pull request by DaveZag**
# Add Integration spec
### On this branch we:
- [x] **Fixed the **n+1** problem**
Before adding `includes`
before
After adding `includes`
after
- [x] **Add test for all the views**

### CODE REVIEW

File: spec/views/user_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    3.times do |a|
+      @user.posts.create(user_id: @user.id, Title: "#{a} Post")
+    end
+  end
+
+  context 'RENDER USER INDEX' do
+    it 'show all content' do
+      visit root_path
+      expect(page).to have_content(@user.Name)
+      # expect(page).to have_xpath("//img[@src=#{@user.Photo}]")
```
Comment (TedLivist) [inline spec/views/user_index_spec.rb]: - [x] Kindly remove commented code.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = Post.where(user_id: @user.id)
+    @posts = Post.includes(comments: [:user]).where(user: @user)
+    # Post.where(user_id: @user.id)
```
Comment (TedLivist) [inline app/controllers/posts_controller.rb]: - [x] Kindly remove commented code.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @DaveZag and @svitalis123,

Good job so far!
✔️ Well-written and passing tests
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by TedLivist** [comment]
## STATUS: INVALID CODE REVIEW REQUEST 🛑 🛑

Hi @DaveZag and @svitalis123,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

### You seem to have submitted your project incorrectly. This project should be a new project in the other partner's repo but you have submitted a request for the first review on the initial repo. Kindly fix that by submitting the initial form in continuation of this project.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: spec/views/post_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    user = User.create(Name: 'John', Photo: 'https://kiddy.com/pic/890987655', Bio: 'Hi there', PostsCounter: 0)
+    Post.create(Title: 'Hello from Mars',
+                Text: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry.', CommentsCounter: 0, LikesCounter: 0, user_id: user.id) # rubocop:disable Layout/LineLength
+    @post = Post.create(Title: 'Hello from Earth', Text: 'The printing and typesetting industry.',
+                        CommentsCounter: 0, LikesCounter: 0, user_id: user.id)
+    @comment = Comment.create(user_id: user.id, post_id: @post.id, Text: 'Hello')
+
+    visit user_posts_path(user.id)
+  end
```
Comment (Davidosky007) [inline spec/views/post_index_spec.rb:15]: * Kindly add an integration test for `I can see the user's profile picture.` as stated in the [project requirement ](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=post%20index%20page%3A-,I%20can%20see%20the%20user%27s%20profile%20picture.,-I%20can%20see)

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @DaveZag @svitalis123 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Good looking readme file 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/post_index_spec.rb (hunk 3 above)
Comment (svitalis123) [reply spec/views/post_index_spec.rb:15]: resolved thank you

**Review by julie-ify — APPROVED** [review]
Hi @DaveZag and @svitalis123,

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**